src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -5207,13 +5207,34 @@ protected sealed override ImmutableArray<ISymbol> GetDeclaredSymbolsCore(SyntaxN
 
             CheckSyntaxNode(node);
 
-            if (node.GetInterceptableNameSyntax() is { } nameSyntax && Compilation.TryGetInterceptor(nameSyntax.GetLocation()) is (_, MethodSymbol interceptor))
+            if (node.GetInterceptableNameSyntax() is { } nameSyntax && Compilation.TryGetInterceptor(nameSyntax) is (_, MethodSymbol interceptor))
             {
                 return interceptor.GetPublicSymbol();
             }
 
             return null;
         }
+
+#pragma warning disable RSEXPERIMENTAL002 // Internal usage of experimental API
+        public InterceptableLocation? GetInterceptableLocation(InvocationExpressionSyntax node, CancellationToken cancellationToken)
+        {
+            CheckSyntaxNode(node);
+            if (node.GetInterceptableNameSyntax() is not { } nameSyntax)
+            {
+                return null;
+            }
+
+            var tree = node.SyntaxTree;
+            var text = tree.GetText(cancellationToken);
+            var path = tree.FilePath;
+            var checksum = text.GetContentHash();
```

Comment:
`GetContentHash` currently uses `XxHash128` but its documentation isn't tied to that specific algorithm. On the other hand, the documentation for `GetInterceptableLocation` promises `XxHash128`.
I'm wondering how to better align this so we don't get a surprise if we ever decide to change `GetContentHash`. 
Should we just remove this this promise from the `GetInterceptableLocation` documentation?

Reply:
> I'm wondering how to better align this so we don't get a surprise if we ever decide to change `GetContentHash`.

Test `Checksum_02` is effectively verifying that the algorithm in use is `XxHash128`. So, a change that just changes the GetContentHash algorithm by itself will start failing tests.

If someone wanted to change `SourceText.GetContentHash()`, they would need to at least retain an internal implementation of `XxHash128` hashing of file content, to allow the compiler to continue to consume existing data in the v1 encoding.

Most likely, we would also introduce a "v2" of the interceptors location encoding at that time, which uses the new, presumably more preferred content hashing algorithm.